### PR TITLE
enhancement #750: Improved updating avatar:

### LINKED ISSUE #750: Improve updating avatars
So, we have endpoint that updates avatar of certain user:

image

You need to improve **2 things**:

1) When updating avatar, a prod link is always generated. You should consider creation of dev link (eg add DEV_BASE_URL env variable).

2) Add a check for existence of avatar link. If it exists, then you should delete it if other users don't have the same link.

### PR DESCRIPTION
**Pull request by JessFreak**
- created FileService.getPathFromLink
- created FileService.checkFileExist
- created FileService.deleteFile
- created UserRepository.findMany
- fixed Swagger documentation

closes #750

### CODE REVIEW

**Review by f-ogoog — APPROVED** [review]

**Comment by ArtemLyah** [comment]
image
Use this documentation style if swagger:
image

**Review by ArtemLyah — APPROVED** [review]

File: src/v2/utils/files/FileService.ts [hunk 1]
```diff
@@ -19,6 +19,26 @@ export class FileService {
     return resolve(process.env.BASE_URL, join(directory, fileName + extname(file.originalname)));
   }
 
+  getFileName (avatar: string) {
+    const parts = avatar.split('/');
+    return parts[parts.length - 1];
+  }
+
+  async checkAvatarExist (avatar: string) {
```
Comment (fokaaas) [inline src/v2/utils/files/FileService.ts:27]: Abstract this method, its task will simply check existence of files depending on whether they are private or static

File: src/v2/utils/files/FileService.ts [hunk 2]
```diff
@@ -19,6 +19,26 @@ export class FileService {
...
+    const filePath = join(__dirname, 'static', 'avatars', avatar);
+    try {
+      await fs.promises.access(filePath);
```
Comment (fokaaas) [inline src/v2/utils/files/FileService.ts:30]: Use fs.existsSync

File: src/v2/utils/files/FileService.ts [hunk 3]
```diff
@@ -19,6 +19,26 @@ export class FileService {
...
+      return true;
+    } catch {
+      return false;
+    }
+  }
+
+  async deleteAvatar (avatar: string) {
```
Comment (fokaaas) [inline src/v2/utils/files/FileService.ts:37]: Abstract this method, its task will simply delete files depending on whether they are private or static

**Review by fokaaas — CHANGES_REQUESTED** [review]

**Review by fokaaas — APPROVED** [review]

**Pull request merged**